app/controllers/comment_controller.rb

```diff
@@ -30,12 +34,51 @@ def create
           end
         end
       end
-    else
+    rescue CommentError
       flash[:error] = 'The comment could not be saved.'
       render text: 'failure'
     end
   end
 
+  def create_by_token
+    @node = Node.find params[:id]
+    @user = Users.find_by_username params[:username]
+    @body = pararms[:body]
+    @token = request.headers["token"]
+
+    if @user && @user.token == @token
+      begin
+        @comment = create_comment(@node, @user, @body)
+        msg = {
```

Comment:
I guess this just has a /lot/ of repeated code here -- the only unique line in 52-59 vs 61-68 vs 71-78 is `:created` and `"Created"`, you know? If you're going through the trouble to a whole new class for CommentError, i just feel that it's not justified given how little this code is reducing repetition. Also, isn't it a bit odd to need to `begin/rescue` in here? 

I'm trying to find a guide to best-practices in Rails controller error handling, to help DRY out this code. But I'm not finding much that's not based on model validation error handling...